### ISSUE TITLE
Hello

### ISSUE THREAD

**Issue by unnellu** [jankovicsandras/imagetracerjs#7]
How about image http://i.imgur.com/Md7ShYi.png , when is convert to svg we have http://i.imgur.com/DvQ0KU6.png , how to remove black inside to tranperent?

**Comment by jankovicsandras** [comment]
Hi,

Can you please give me more details on the issue? What ```options``` did you use? Have you tried to convert multiple times? I recommend reading [Options for deterministic tracing](https://github.com/jankovicsandras/imagetracerjs/blob/master/deterministic.md) and maybe using a custom palette.

**Comment by unnellu** [comment]
var myImageData = { 'width':512, 'height':512, 'data':png.data };
		
        // tracing
        var options = {"blurdelta":0,"blurradius":0,"colorquantcycles":2,"colorsampling":null,"desc":false,"lcpr":null,"ltres":0.5,"mincolorratio":0.02,"numberofcolors":256,"pathomit":1,"qcpr":null,"qtres":1,"roundcoords":1,"scale":1,"simplifytolerance":0,"viewbox":false}; // optional

**Comment by unnellu** [comment]
i use nodejs, but this problem i see in browser conver with javascript

**Comment by jankovicsandras** [comment]
I recommend changing some options, especially setting ```colorsampling:true```. (You can also omit options with default values. E.g. blur is already off by default, you don't need to set ```blurradius``` and ```blurdelta``` to ```0```.) 

Please try it like this and run it several times:

```javascript
var options = {
    "colorquantcycles":2,
    "colorsampling":true, // I recommend true (or false but not null)
    "desc":false,
    "ltres":0.5,
    "mincolorratio":0.02,
    "numberofcolors":256,
    "pathomit":1,
    "qtres":1,
    "roundcoords":1
}; 

// If you want deterministic results, please define a custom palette
// and set colorquantcycles:1

options.colorquantcycles = 1;

options.pal = [
{r:0,g:0,b:0,a:0}, // alfa = 0, transparent color 
{r:0,g:0,b:255,a:255}, 
// ... so many colors as you want
{r:255,g:255,b:0,a:255}
];

```

Please let me know if this works or not! :)

**Comment by jankovicsandras** [comment]
As I see transparency works with ```colorsampling:true```, but it creates too big files. Here are 2 ideas you can try:

1. ```colorsampling:true``` but with less colors, e.g. ```numberofcolors:64``` or ```numberofcolors:32```

2. ```colorsampling:false``` , generating a custom palette manually, and then setting a transparent color manually

```javascript
var options = {
    "colorquantcycles":2,
    "colorsampling":false,
    "desc":false,
    "ltres":0.5,
    "mincolorratio":0.02,
    "numberofcolors":256,
    "pathomit":1,
    "qtres":1,
    "roundcoords":1
}; 

// Generating a custom palette
options.pal = ImageTracer.generatepalette(256);

// Setting the last color to transparent manually
options.pal[255] = { r:0, g:0, b:0, a:0 };
```

I haven't tested these (sorry). Let me know if these work or not.

**Comment by unnellu** [comment]
transparent  work only with 
`var options = {
    "colorquantcycles":2,
    "colorsampling":true, // I recommend true (or false but not null)
    "desc":false,
    "ltres":0.5,
    "mincolorratio":0.02,
    "numberofcolors":256,
    "pathomit":1,
    "qtres":1,
    "roundcoords":1
}; `
Not work with 
// Generating a custom palette
options.pal = ImageTracer.generatepalette(256);

// Setting the last color to transparent manually
options.pal[255] = { r:0, g:0, b:0, a:0 };
or 
numberofcolors:64 
or 
numberofcolors:32

For example i did simple covertor 
http://lp.kynoty.ru/7/index.html

**Comment by unnellu** [comment]
maybe  i can help? upload  ***/7/index.html all files in zip arhive to you ? or write another script with all paramentr in option?

**Comment by unnellu** [comment]
In this example really not work http://lp.kynoty.ru/7/index2.html

**Comment by unnellu** [comment]
image http://lp.kynoty.ru/7/img/efefs.png

**Comment by jankovicsandras** [comment]
Sorry, I was on holiday. I will test your examples and reply as soon as possible.

**Comment by unnellu** [comment]
Hello, and how to fix the problem?

**Comment by unnellu** [comment]
Thanks

**Comment by unnellu** [comment]
Hello, maybe you can  do it for me? I can pay for it, can you remove this mistake?

**Comment by jankovicsandras** [comment]
Transparent hole issue is fixed in version 1.2.0.

**Comment by unnellu** [comment]
thanks
